pkg/asset/installconfig/azure/validation.go

```diff
@@ -77,6 +77,11 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUsAvailable", req.minimumVCpus)
 				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
 			}
+		} else if strings.EqualFold(*capability.Name, "HyperVGenerations") {
+			generation := to.String(capability.Value)
+			if !strings.Contains(generation, "V1") {
```

Comment:
This probably won't ever bite us, but this is fragile to generations greater than 9.
```suggestion
			if !sets.NewString(strings.Split(generation, ",")...).Has("V1")
```
https://go.dev/play/p/xkPDVfR9kK0

For completeness, it is probably worthwhile to convert `generation` to upper-case and trim whitespace around the generations, too.